enterprise/reporting/util_test.go

```diff
@@ -0,0 +1,818 @@
+				StatusDetail: &types.StatusDetail{
+					SampleEvent: nil,
+				},
+			},
+		},
+		{
+			name: "Empty sample event",
+			metric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent: emptySampleEvent,
+				},
+			},
+			wantMetric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent: emptySampleEvent,
+				},
+			},
+		},
+		{
+			name: "Event Sampler returns get error",
+			metric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			wantMetric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			shouldGet: true,
+			getError:  errors.New("get error"),
+		},
+		{
+			name: "Event Sampler returns put error",
+			metric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			wantMetric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			shouldGet: true,
+			shouldPut: true,
+			putError:  errors.New("put error"),
+		},
+		{
+			name: "Sample is not found",
+			metric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			wantMetric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			shouldGet: true,
+			shouldPut: true,
+		},
+		{
+			name: "Sample is found",
+			metric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    sampleEvent,
+					SampleResponse: sampleResponse,
+				},
+			},
+			wantMetric: types.PUReportedMetric{
+				StatusDetail: &types.StatusDetail{
+					SampleEvent:    emptySampleEvent,
+					SampleResponse: emptySampleResponse,
+				},
+			},
+			shouldGet: true,
+			found:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockEventSampler := mockEventSampler.NewMockEventSampler(ctrl)
+			if tt.shouldGet {
+				mockEventSampler.EXPECT().Get(gomock.Any()).Return(tt.found, tt.getError)
+			}
+			if tt.shouldPut {
+				mockEventSampler.EXPECT().Put(gomock.Any()).Return(tt.putError)
+			}
+
+			gotMetric, err := transformMetricWithEventSampling(tt.metric, 1234567890, mockEventSampler, 60)
+			if tt.getError != nil || tt.putError != nil {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+			require.Equal(t, tt.wantMetric.StatusDetail.SampleEvent, gotMetric.StatusDetail.SampleEvent)
+			require.Equal(t, tt.wantMetric.StatusDetail.SampleResponse, gotMetric.StatusDetail.SampleResponse)
+		})
+	}
+}
```

Comment:
@koladilip Why are we moving the tests from `reporting_test.go` to `utils_test.go` ? Most of it is testing code `reporting.go`

Reply:
Can we move `TestGetAggregatedReports()` to `reporting_test.go` ?

Reply:
getAggregationBucketMinute function is defined in utils.go so maybe it is easy to maintain in utils_test.go